src/pybind/mgr/dashboard/controllers/rgw.py

```diff
@@ -121,3 +122,23 @@ def create(self, bucket, uid):
             cherrypy.response.headers['Content-Type'] = 'application/json'
             cherrypy.response.status = 500
             return {'detail': str(e)}
+
+
+@ApiController('/rgw/user')
+@AuthRequired()
+class RgwUser(RESTController):
+
+    def delete(self, uid):
+        try:
+            rgw_client = RgwClient.admin_instance()
+
+            # Ensure the user is not configured to access the Object Gateway.
+            if rgw_client._SYSTEM_USERID == uid:
```

Comment:
While testing this PR I found that I'm still able to delete the system user.

On my environment, `rgw_client._SYSTEM_USERID` is an empty string, so it's different than `uid`.

Note that I've only executed the required commands:

 - `ceph dashboard set-rgw-api-access-key <access_key>`
 -  `ceph dashboard set-rgw-api-secret-key <secret_key>`.

I've **NOT** executed `ceph dashboard set-rgw-api-user-id <user_id>` since it's optional.

Reply:
That is exactly the problem, the user ID must be a required config setting. Without this information it is not possible to do this check.

If the user ID is not set a possible solution might be to find the user ID via enumerating all users and their access/secret while initializing the [instance of RgwClient](https://github.com/ceph/ceph/blob/master/src/pybind/mgr/dashboard/services/rgw_client.py#L119).

Reply:
Another solution might be to request the user info (incl. keys) of the user to delete before triggering the deletion. Thus we can check if the configured access/secret key is owned by the user to be deleted. This would only require one additional RGW AdminOPS API call.

Reply:
@votdev why not access `/metadata/user` the first time we open the connection and get the user id of the current user?

Reply:
@rjfd You're right, i missed that possibility.

Reply:
@rjfd /metadata/user returns a list of ALL users but not the user data incl. the keys. So this does not help.